### DISCUSSION TITLE
Provide dependency globally?

### DISCUSSION THREAD

**Discussion by PatrikBird** [vitest-dev/vitest#818] [Q&A] [answered]
Hey guys! I started playing around with Vitest and I really like it so far, **thanks for creating such an amazing tool!** 

I'm currently migrating a Project that uses [msal-browser](https://www.npmjs.com/package/@azure/msal-browser) for authentication.
This dependency relies on crypto APIs which are not available while running the tests, hence I need to provide them.

Using Jest, I could do this globally as shown [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/1840#issuecomment-664985444). 

Is there an equivalent in Vitest to this? Or should I use the [globalSetup](https://vitest.dev/config/#globalsetup)?

Cheers! Patrik

**Comment by sheremet-va** [comment]
Don't think `globalSetup` will work, since it is run in the main thread, and not in tests
You can just import crypto in [`setupFiles`](https://vitest.dev/config/#setupfiles) and assign it to `globalThis`:
```ts
import crypto from 'crypto'
globalThis.crypto = crypto
```

**Reply by PatrikBird** [reply]
thanks for your quick response. I’ll test this tonight

**Reply by MykalMachon** [reply]
@sheremet-va Thank you 🙏🏻
This worked for a similar issue I was having! lifesaver.

**Reply by PatrikBird** [reply]
Sorry for my late reply. With 
```js
globalThis.crypto = crypto
```
in my setupFile.ts, I'm getting the following type error:
`Type 'typeof import("crypto")' is missing the following properties from type 'Crypto': subtle, getRandomValuests(2739)` 

If I just ignore it, the tests still fail with the same `crypto_nonexistent` error as before.

Edit: I don't expect to get the solution served on a silver platter. Just FYI 😄

**Reply by PatrikBird** [reply]
It worked with
```js
globalThis.window.crypto = crypto;
```
Just the type error remains, which is no biggie. Thank you!

**Comment by PatrikBird** [comment, marked as answer]
Add the following using [setupFiles](https://vitest.dev/config/#setupfiles):
```js
import crypto from 'crypto'
globalThis.window.crypto = crypto;
```
